spec/source_file_spec.rb

```diff
@@ -80,8 +113,38 @@
       end
     end
 
+    context "A file that have inline branches" do
+      COVERAGE_FOR_DUMB_INLINE = {
+        :lines => [nil, 1, 1, 1, nil, nil, 1, 0, nil, nil, nil, nil, nil, nil, nil, nil],
+        :branches => {[:if, 0, 18, 6, 18, 9] => {[:then, 1, 18, 8, 18, 81] => 3, [:else, 2, 18, 8, 19, 19] => 0}, [:if, 3, 29, 6, 35, 9] => {[:then, 4, 30, 8, 30, 81] => 3, [:else, 5, 31, 8, 34, 20] => 0}},
+      }.freeze
+
+      subject do
+        SimpleCov::SourceFile.new(source_fixture("never.rb"), COVERAGE_FOR_DUMB_INLINE)
+      end
+
+      it "Has branches report on 3 lines " do
+        expect(subject.branches_report.keys.size).to eq(3)
+        expect(subject.branches_report.keys).to eq([18, 29, 30])
+      end
+
+      it "Has covered branches count 2 " do
+        expect(subject.covered_branches.size).to eq(2)
+      end
+
+      it "Has dual element in condition at line 18 report" do
+        expect(subject.branches_report[18]).to eq([[3, "+"], [0, "-"]])
+      end
+
+      it "Has branches coverage precent 50.00" do
+        expect(subject.branches_coverage_precent).to eq(50.00)
```

Comment:
Why not `50.0`?

Reply:
I just followed the standard definition of percentage for line coverage calculation.